tests/main/snapshot-exclusions-dynamic/task.yaml

```diff
@@ -0,0 +1,132 @@
+summary: Check that the basic dynamic snapshot data exclusion functionality is ok
+
+environment:
+    # tar invoked when creating snapshots triggers OOM
+    SNAPD_NO_MEMORY_LIMIT: 1
+    USER_DATA_DIR: /root/snap/test-snap
+    SYSTEM_DATA_DIR: /var/snap/test-snap
+
+prepare: |
+    "$TESTSTOOLS"/snaps-state install-local test-snap
+
+restore: |
+    rm -f expected_options_entry actual_options_entry expected_restored_files actual_restored_files ./*_export.snapshot
+
+debug: |
+    snap saved || true
+
+execute: |
+    # waitAsyncChangeDone() waits for completion of async change
```

Comment:
I think you could use the retry tool instead of defining a new function:
You can try this
retry -n 300 --wait 0.1 sh -c "curl -s --unix-socket /run/snapd.socket http://localhost/v2/changes/\"$change\" | jq '.result.tasks[].progress.done' | NOMATCH 1"

Reply:
Thanks, I used your suggestion